### PR TITLE
add metrics

### PR DESCRIPTION
**Pull request by whi-tw**
What
----

Add metrics endpoints to billing apps

Add metrics proxy app:
- Has `http_sd` endpoint for discovering metrics of other apps
- Has proxy to retrieve metrics from a specific instance

How to review
-----

- Code review
- Ensure tests pass
- Talk to Tom?

Who can review
-----

Not me

---

🚨⚠️ Please do not merge this pull request via the GitHub UI ⚠️🚨

### CODE REVIEW

**Comment by risicle** [comment]
Are you not able to squash down the commits _a bit_? Nobody can sensibly review 48 commits.

**Comment by AP-Hunt** [comment]
Heads up @whi-tw, the tests are failing at the tip of this branch

**Review by malcgds — APPROVED** [review]

File: Makefile
Comment (malcgds) [inline Makefile:1]: gherkin tests have the prod regions mixed up

File: Makefile [hunk 1]
```diff
@@ -1,41 +1,99 @@
...
-# .PHONY: gherkin_test
+
+.PHONY: gherkin_test
 gherkin_test: gherkin_test_lon gherkin_test_ie
 
-# .PHONY: gherkin_test_lon
+.PHONY: gherkin_test_lon
 gherkin_test_lon:
 	mkdir -p gherkin/features
 	cp ../paas-cf/config/billing/tests/eu-west-1_billing_rds_charges.feature gherkin/features/
```
Comment (malcgds) [inline Makefile:99]: gherkin tests have the prod regions mixed up but appears as though prior to current changes

**Pull request merged**